src/azure-cli-core/azure/cli/core/sdk/policies.py

```diff
@@ -0,0 +1,98 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import logging
+import re
+import types
+
+from azure.core.pipeline.policies import SansIOHTTPPolicy
+from knack.log import get_logger
+
+_LOGGER = get_logger(__name__)
+
+
+class SafeNetworkTraceLoggingPolicy(SansIOHTTPPolicy):
+    """The logging policy that redacts specified headers.
+    Based on azure.core.pipeline.policies._universal.NetworkTraceLoggingPolicy
+    """
+
+    def __init__(self, headers_to_redact=None):
+        """
+        :param list[str] headers_to_redact: headers that should be redacted from the log.
+          Default to 'Authorization', 'x-ms-authorization-auxiliary'.
+        """
+        if headers_to_redact is not None:
+            self.headers_to_redact = headers_to_redact
+        else:
+            self.headers_to_redact = ['authorization', 'x-ms-authorization-auxiliary']
+
+    def on_request(self, request):
+        http_request = request.http_request
+        options = request.context.options
+        logging_enable = options.pop("logging_enable", True)
+        request.context["logging_enable"] = logging_enable
+        if logging_enable:
+            if not _LOGGER.isEnabledFor(logging.DEBUG):
+                return
+
+            try:
+                _LOGGER.debug("Request URL: %r", http_request.url)
+                _LOGGER.debug("Request method: %r", http_request.method)
+                _LOGGER.debug("Request headers:")
+                for header, value in http_request.headers.items():
+                    if header.lower() in self.headers_to_redact:
+                        value = '*****'
+                    _LOGGER.debug("    %r: %r", header, value)
+                _LOGGER.debug("Request body:")
+
+                # We don't want to log the binary data of a file upload.
+                if isinstance(http_request.body, types.GeneratorType):
+                    _LOGGER.debug("File upload")
+                    return
+                try:
+                    if isinstance(http_request.body, types.AsyncGeneratorType):
+                        _LOGGER.debug("File upload")
+                        return
```

Comment:
In storage track1 data plane SDK, for such file, it will logging with file size. Could we also support it here?

Reply:
This `SafeNetworkTraceLoggingPolicy` is designed as a generic policy. That's why I am hesitating to apply it to all SDKs (including data-plane SDKs) at the beginning.

If storage data-plane SDK or other data-plane SDKs requires additional/special logic, we may apply `SafeNetworkTraceLoggingPolicy` only to ARM and let data-plane SDKs decide what policy they want.

Another solution is to define your own policy and override 

https://github.com/Azure/azure-cli/blob/36e3d156e14f2260d1b4a5281adffcdb1446384e/src/azure-cli-core/azure/cli/core/commands/client_factory.py#L155-L156

Reply:
We could keep current design then.